### PR TITLE
Do not reset spree preferences starting from v2.9

### PR DESCRIPTION
**Pull request by kennyadsl**
We need this to remove deprecation warnings into extensions. Of course, this could let some spec fail on 2.9 or master.

This PR stops resetting preferences after each spec run and provides an alternative method to use into extensions (and stores) specs. This method will work for all supported Solidus versions and it will be easy to remove later since they have the same signature: we can just remove the `require 'solidus_support/testing_support/preferences'` for solidus support `rails_helper` without any change in the extensions.

### CODE REVIEW

**Comment by kennyadsl** [comment]
@spaghetticode can you please review?

**Review by spaghetticode — APPROVED** [review]
LGTM 🍸 
thank you @kennyadsl 👍

**Comment by kennyadsl** [comment]
Better to wait for https://github.com/solidusio/solidus/pull/3275, which could change the signature of the `stub_spree_preferences` method.

**Review by ericsaupe — APPROVED** [review]
LGTM!

**Comment by kennyadsl** [comment]
I tried this branch in a real extension (https://github.com/solidusio/solidus_auth_devise/pull/166), waiting for the CI to run before merge here.

**Pull request merged**
